ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -163,7 +163,15 @@ public static boolean mustBeMaterializedForTableJoins(final ColumnName columnNam
   }
 
   public static int getPseudoColumnVersionFromConfig(final KsqlConfig ksqlConfig) {
+    return getPseudoColumnVersionFromConfig(ksqlConfig, false);
+  }
+
+  public static int getPseudoColumnVersionFromConfig(
+      final KsqlConfig ksqlConfig,
+      final boolean forPullOrScalablePushQuery
+  ) {
     return ksqlConfig.getBoolean(KsqlConfig.KSQL_ROWPARTITION_ROWOFFSET_ENABLED)
+        && !forPullOrScalablePushQuery
```

Comment:
Hacking around the pseudocolumn version in this way feels brittle. We should update this to use the new `SystemColumns#isDisallowedInPullQueries()` method being added in https://github.com/confluentinc/ksql/pull/8135. I'm fine with merging this PR first and having @Sullivan-Patrick update this in his PR in order to unblock the cyclic dependency, but I definitely don't think we should hacking pull queries to use the old pseudocolumn version as in this PR when the feature goes out.

Reply:
I agree.  This part does feel a little brittle and was a bit of a stab at getting it working, but I agree I like that method better.  Happy to let him resolve it if that works.